database/mysql_anonymiser.go

```diff
@@ -3,15 +3,15 @@ package database
 import (
 	"database/sql"
 	"fmt"
-	"log"
 	"reflect"
 	"strings"
 
 	"github.com/hellofresh/klepto/utils"
 	"github.com/spf13/viper"
 )
 
-const literalPrefix = "literal:"
+// LiteralPrefix defines the constant we use to prefix literals
+const LiteralPrefix = "literal:"
 
 // MySQLAnonymiser anonymises MySQL tables and
 type MySQLAnonymiser struct {
```

Comment:
ATM `MySQLAnonymiser` has some side effects during db calls e.g `a.conn.Query` mixed with some pure functions `shouldNotAnonymise , FindsFaker` and etc...

In order to make the code testable, let's separate the concerns a bit and rely on dependency injection,

Rather than `MySQLAnonymiser` be receiving a concrete db connection implementation, let's expose `Datastore` and receive it via constructor, this way we can create stubs for datastore and test the pure function we've got on MySQLAnonymiser only via the public API which is `DumpTable(table string, rowChan chan<- []*Cell, endChan chan<- bool) error`

Show me code 👨‍💻 

```go
type Datastore interface {
  All(table string) (*Rows, error)
}

// implement datastore somewhere

type MySQLAnonymiser struct {
  typeScanner utils.TypeScanner
  datastore      Datastore
}

// public API
func (m *MySQLAnonymiser) DumpTable(table string, rowChan chan<- []*Cell, endChan chan<- bool) error {
  rows, err := m.datastore.All(table)
  // transform stuff
}
```

In orde to test you can create a struct which implement the `Datastore` methods, e.g.
```go
type DatastoreStub struct {}

func (ds *DatastoreStub) All(table string) (*Rows, error) {
  return &Rows{
    // fill Rows struct with data you need
  }
}
```

WDYT?

Reply:
If you need let's pair on it, let's just schedule some time 😉

Reply:
I like this a lot :)

Reply:
Sounds really great! I got started on refactoring to separate out the concerns a little bit. DumpTable is certainly the main piece! So will get onto breaking that into smaller pieces using your suggestions. And of course, if I get stuck on the way, I know where to find you :)